guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
 
-```bash
-$ bin/importmap unpin react
-Unpinning "react"
-Unpinning "object-assign"
-```
+Fingerprinting is a technique that makes the name of a file dependent on the
+contents of the file. A digest of the content is generated and appended to the
+name so that when the file contents changes, so does the digest, and therefore
+the filename is also changed. For static or infrequently changed content, this
+provides an easy way to tell whether two versions of a file are identical, even
+across different servers or deployment dates.
 
-You can control the environment of the package for packages with separate "production" (the default) and "development" builds:
+When a filename is unique and based on its content, HTTP headers can be set to
+encourage caches everywhere (whether at CDNs, at ISPs, in networking equipment,
+or in web browsers) to keep their own copy of the content. When the content is
+updated, the fingerprint will change. This will cause the remote clients to
+request a new copy of the content. This is generally known as _cache busting_.
 
-```bash
-$ bin/importmap pin react --env development
-Pinning "react" to https://ga.jspm.io/npm:react@17.0.2/dev.index.js
-Pinning "object-assign" to https://ga.jspm.io/npm:object-assign@4.1.1/index.js
-```
+When fingerprinted, a file like `styles.css` is renamed to
+`styles-a1b2c3d4e5f6.css`. This ensures that if `styles.css` is updated, the
+filename changes as well, compelling the browser to download the latest version
+instead of using a potentially outdated cached copy.
 
-You can also pick an alternative, supported CDN provider when pinning, like [`unpkg`](https://unpkg.com/) or [`jsdelivr`](https://www.jsdelivr.com/) ([`jspm`](https://jspm.org/) is the default):
+A manifest is a file that maps logical asset paths to their corresponding
+fingerprinted filenames. The manifest is used to look up the correct filename
+for an asset. The manifest is typically a JSON file that looks like this:
 
-```bash
-$ bin/importmap pin react --from jsdelivr
-Pinning "react" to https://cdn.jsdelivr.net/npm/react@17.0.2/index.js
+```json
+{
+  "application.js": "application-1a2b3c4d5e6f.js",
+  "application.css": "application-1a2b3c4d5e6f.css"
+}
 ```
 
-Remember, though, that if you switch a pin from one provider to another, you may have to clean up dependencies added by the first provider that isn't used by the second provider.
-
-Run `bin/importmap` to see all options.
-
-Note that this command is merely a convenience wrapper to resolving logical package names to CDN URLs. You can also just lookup the CDN URLs yourself, and then pin those. For example, if you wanted to use Skypack for React, you could just add the following to `config/importmap.rb`:
+Fingerprinting is enabled by default for both the development and production
+environments. When you reference an asset in your application, Rails uses the
+manifest to find the correct fingerprinted filename to include in the HTML
+output.
+
+Features
```

Comment:
Is it accurate to call these Propshaft features?

```suggestion
Propshaft Features
```

Reply:
Yes, I think this makes sense.